cpp/src/arrow/scalar_test.cc

```diff
@@ -1984,4 +1985,41 @@ TEST_F(TestExtensionScalar, ValidateErrors) {
   AssertValidationFails(scalar);
 }
 
+template <typename T>
+class TestScalarScratchSpace : public ::testing::Test {
+ public:
+  TestScalarScratchSpace() = default;
+};
+
+TYPED_TEST_SUITE(TestScalarScratchSpace, BaseBinaryOrBinaryViewLikeArrowTypes);
```

Comment:
This doesn't touch the List/ListView cases, please also test those

Reply:
Sure. I was planning to do so before turning this PR to ready status.